#CP-27 Setup Test with Jest & Enzyme

Abdulkeza opened the pull request:
# What does this PR do?
- [x] Configured app to use Jest & enzyme for testing before deploying codes

# Description of Task to be completed?
- [x] configured Jest
- [x] configured enzyme

# How should this be manually tested?
-[Clone](https://github.com/atlp-rwanda/phantom-fe-codebandits)
-npm install
-npm run test  or npm test -- --coverage

# Any background context you want to provide?
(N/A)

# What are the relevant pivotal tracker stories?
[#CP-27](https://trello.com/c/rcRIA4dN/27-setup-continuous-integration-quality-assurance-pipeline)

# Questions:
(N/A)

Abdulkeza committed:
-#CP-27 Setup Test with Jest & Enzyme

Abdulkeza committed:
-#CP-27 Setup Test with Jest & Enzyme

feyton commented on package.json:
```diff
@@ -6,8 +6,22 @@
     "start": "webpack serve",
-    "build": "webpack"
+    "build": "webpack",
+    "test": "jest"
```
It would be better to add the coverage script to the project too. 
like: 
```
"coverage": "script......"
```

Jkadhuwa commented on src/App.js:
```diff
@@ -10,4 +10,4 @@ const App = ()=> {
 }
 
-export default App
+export { App };
```
Why change from default export to named export? You are breaking other people's codes.

feyton replied:
Resolved. The issue was related to the use of incompatible Adapter

Jkadhuwa commented on tests/App.test.js:
```diff
@@ -0,0 +1,16 @@
+        const wrapper = shallow(<App/>);
+        expect(wrapper.find("p").text()).toEqual("powered by CodeBandits");
+    });
+});
```
Always leave a line after your last line of code.

Jkadhuwa commented on tests/setupTests.js:
```diff
@@ -0,0 +1,4 @@
+import Enzyme from 'enzyme';
```
You can destructure configure from enzyme directly. import {configure} from 'enzyme'

feyton replied:
Used the de-structured version of configure.

Jkadhuwa reviewed (changes requested):
Good job, have a look at the feedback left below.

Abdulkeza committed:
coverage added in package.json
export {App} modified into export default App
space added between Lines

Abdulkeza committed:
coverage added in package.json
export {App} modified into export default App
space added between Lines

Abdulkeza committed:
modified

Abdulkeza committed:
modified

feyton committed:
ch-(setup test)- Add the coverage script

The script generate report that are used to generate
code climate report on coverage

feyton reviewed (dismissed):
All test run successfully and generate a report that can be integrated into code climate

Abdulkeza committed:
Test for Redux

feyton reviewed (dismissed):
Further tests will be added after the integration of CI and fix of the redux errors when running tests.

Abdulkeza committed:
.

Abdulkeza committed:
Tests modified

Abdulkeza committed:
Test Modified

Abdulkeza committed:
codeclimate.yml file added

Abdulkeza committed:
codeclimate.yml file modified

Abdulkeza committed:
codeclimate.yml file modified

Abdulkeza committed:
codeclimate.yml file modified

feyton committed:
Remove coverage files from tracked folder

This commit removes the previously committed
folder which is auto generated

feyton reviewed (approved):
All tests are passing and the scripts to generate reports are well integrated

samyshyaka commented on tests/App.test.js:
```diff
@@ -0,0 +1,54 @@
+import { Provider } from 'react-redux';
+import { configureStore } from '@reduxjs/toolkit';
+
+import App from '../src/App';
```
Make sure your code is up to date with the develop branch. You shouldn't have been able to commit without the extensions on your imports.

samyshyaka commented on tests/App.test.js:
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { mount, render, shallow } from 'enzyme';
```
Some of your imports are not being used. I think it's better to remove them

Abdulkeza replied:
okay, i can remove it

samyshyaka commented on package.json:
```diff
@@ -31,8 +51,13 @@
     "babel-loader": "^8.2.3",
     "css-loader": "^6.6.0",
+    "enzyme": "^3.11.0",
+    "enzyme-adapter-react-16": "^1.15.6",
```
Is this adapter being used?

Abdulkeza replied:
i have to remove it, thanks

samyshyaka reviewed (commented):
Nice work so far abdul
